nzediegwu1/blog-connect-web
# #162652884 Dockerize app for development and production

#### What does this PR do?
Enable a developer to create and run docker image of the application for seamless development and production purpose

#### Description of Task to be completed?
- create .dockerignore file for files to be ignored when building image
- create Dockerfile to define commands to execute for common image
- create docker-compose file to define build process for development
- install pushstate-server to serve react-app on production
- define production start script `start:prod` in package.json

#### How should this be manually tested?
- When a developer runs command: `docker-compose up` on terminal, the application should be auto built and started in development
- When a developer makes changes in the code, dev server should automatically be reloaded to ensure changes are reflected
- Developer should be able to run command: `docker build -t blog-connect-web .` to build the project for production, and `docker run -p 9000:9000 blog-connect-web` to start the production build of the react application



#### Any background context you want to provide?
When the application is dockerized, the stress required to install dependencies and setup the project on local machine is eliminated. Enabling a new developer to focus on writing code than setting up project

#### What are the relevant pivotal tracker stories?
[#162652884](https://www.pivotaltracker.com/story/show/162652884)

#### Screenshots (If applicable)
N/A
